gammapy/datasets/map.py

```diff
@@ -169,13 +197,121 @@ def _default_binsz(observation, spatial_bin_size_min=0.01 * u.deg):
 
 def _default_width(observation, spatial_width_max=12 * u.deg):
     # width estimated from the rad_max or the offset_max
-    if observation.rad_max is not None:
+    if hasattr(observation, "rad_max") and observation.rad_max is not None:
         width = 2.0 * np.max(observation.rad_max)
-    else:
+    elif hasattr(observation.psf, "axes"):
         width = 2.0 * observation.psf.axes["offset"].edges[-1]
+    else:
+        width = 2.0 * np.max(observation.psf.psf_map.geom.width)
     return np.minimum(width, spatial_width_max)
 
 
+def create_empty_map_dataset_from_irfs(
```

Comment:
I think the function is doing too much which makes it complex to understand.
- it can create a `MapDataset` without any IRF, contrary to its name, if optional arguments are all passed. This use case is covered by `MapDataset.create`.
- it deals with DL3 and DL4 IRFs at the same time. I think those are different use cases. The latter is closer to a `MapDataset.copy()`

Reply:
removed the support for MapDatasest but kept the support for DL4 irfs